src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -505,6 +528,76 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
     }
   }
 
+  override def checkGoogleGroupExists(groupEmail: String): Boolean = {
+    val directoryService = getDirectoryManager(getDelegatedCredentialForAdminUser)
+    val checkGroupRequest = directoryService.groups.get(groupEmail)
+    Try(executeGoogleRequest(checkGroupRequest)) match {
+      case Failure(_) => false
```

Comment:
There could be a variety of failure modes here. It might make sense to check the response code and maybe log errors that are unexpected. Not having looked at the APIs, I would expect that anything other than a 404 would be a good candidate to log.

Reply:
I changed this function to `checkUserIsInExistingGoogleGroup` and return the error code (and log non-404 failures) or 200 if successful. a 404 response results (logic in `UserService.scala`) prompts to create a brand new google group to avoid overwriting another one, and if there's a different kind of error, we do nothing.